hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/ratis/OzoneManagerDoubleBuffer.java

```diff
@@ -173,17 +196,19 @@ private void flushTransactions() {
                 flushedTransactionsSize);
           }
 
-          long lastRatisTransactionIndex =
-              readyBuffer.stream().map(DoubleBufferEntry::getTrxLogIndex)
-                  .max(Long::compareTo).get();
-
-          List<Long> flushedEpochs =
-              readyBuffer.stream().map(DoubleBufferEntry::getTrxLogIndex)
-                  .sorted().collect(Collectors.toList());
+          // When non-HA do this step here, as the sorted is not require in
+          // flush to DB. As in non-HA We want to complete futures as quick
+          // as possible after flush to DB, to release rpc handler threads.
+          if (!isRatisEnabled) {
```

Comment:
Sorry this comment is not clear. Do you mean sorted Transaction list is not required here?

Reply:
Yes, at this point in time we don't require the list to be sorted for non-HA, the sorted list is required for cleanupCache only, so delaying this. The reason for this already explained in the comment.

Reply:
Time for this sorting should be inconsequential. The readyBuffer would have 10s of transactions at max. 
Do we really need to separate it for HA and non-HA? 
NIT: Can you please fix the typo require -> required and capital W in We.

Reply:
Some time, we see this list grow up to 100-300. And for non-HA, all rpc handlers are blocked on flush futures, intention here is does not want to spend time on unnecessary operations and make the flush futures complete and unblock rpc handlers.